account_banking_pain_base/models/account_payment_order.py

```diff
@@ -487,9 +487,16 @@ def generate_party_block(
 
     @api.model
     def generate_remittance_info_block(self, parent_node, line, gen_args):
-        remittance_info = etree.SubElement(
-            parent_node, 'RmtInf')
-        if line.communication_type == 'normal':
+
+        lsv_plus = False
+        if parent_node.tag != "RmtInf":
+            remittance_info = etree.SubElement(
+                parent_node, 'RmtInf')
+        else:
+            remittance_info = parent_node
+            lsv_plus = True
+
+        if line.communication_type == 'normal' and not lsv_plus:
```

Comment:
Why not inheriting `generate_remittance_info_block` and change `creditor_reference.text` value after calling super() and checking your are in the case of LSV+ ? I still think you are not obliged to changed the code of this module.